base/MaplGrid.F90

```diff
@@ -325,6 +329,27 @@ subroutine MAPL_GridGet(GRID, globalCellCountPerDim, localCellCountPerDim, RC)
          end if
       end if
 
+      if (present(layout)) then
+         call ESMF_GridGet(grid,distgrid=distgrid,dimCount=dimCount,_RC)
+         call ESMF_DistGridGet(distgrid,deCount=deCount,_RC)
+         allocate(minindex(dimCount,decount),maxindex(dimCount,decount))
+
+         call MAPL_DistGridGet(distgrid, &
+            minIndex=minindex, &
+            maxIndex=maxindex, _RC)
+         if (associated(ims)) then
+            nullify(ims)
+         end if
+         if (associated(jms)) then
+            nullify(jms)
+         end if
```

Comment:
No need to be so verbose here.  (I had to check/learn that nullify accepts multiple items.)
```suggestion
         nullify(ims, jms)
```

Reply:
And as per Atanas' comment below - is this a memory leak?

Reply:
@tclune  I don't know what is going on. That seems to be necessary to make gfortran happy since it complains IMS is associated going in the call. If I try deallocating it rather than nullifying it it says it can't deallocate something not allocated.
Really there's no reason that ims/jms passed needs to be a pointer at all from what I can see. Seems like it could be turned into an allocatable. Unfortunately that call is used elsewhere so I would have to fix every there it is called to make it an allocatable rather than a pointer?
Maybe I should?

Reply:
2 separate issues.   **If** the pointers do not "own"  the memory, then my suggested change is just simpler.  Than your original.  The pointers end up null regardless of their state.

But **if** the pointers do "own" the memory, then we should/must deallocate.  Then the code should be:
```
if (associated(ims)) then
   deallocate(ims)
endif
```
Then `ims` ends up in null() in any event and any associated memory is freed.  I'll try to refresh my memory to see which case should apply.